services/provider/server/server.go

```diff
@@ -1042,3 +1043,143 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) GetStorageClientsInfo(ctx context.Context, req *pb.StorageClientsInfoRequest) (*pb.StorageClientsInfoResponse, error) {
+	var clientsInfo []*pb.ClientInfo
+	var clientError []*pb.StorageClientInfoError
+
+	for i := range req.ClientIDs {
+		consumer, err := s.consumerManager.Get(ctx, req.ClientIDs[i])
+		if err != nil {
+			klog.Errorf("failed to get consumer with id %v: %v", req.ClientIDs[i], err)
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  err.Error(),
+				},
+			)
+			continue
+		}
+		owner := util.FindOwnerRefByKind(consumer, "StorageCluster")
+
+		if owner == nil {
+			klog.Errorf("failed to find StorgeCluster owning the StorageConsumer %v", req.ClientIDs[i])
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to find StorgeCluster owning the StorageConsumer",
+				},
+			)
+			continue
+		}
```

Comment:
Why is this an error? this is equivalent to not found

Reply:
This is an error because for the clientID in the req there was no storageCluster owning the consumer. So I am aggregating it into the error Info struct and then sending it back at the end. It will continue reconcile for other clientIDs